### PR TITLE
[receiver/statsdreceiver] update to use generated status

### PR DESCRIPTION
**Pull request by my-git9**
**Description:** <Describe what has changed.>

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21213
**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by dmitryax — APPROVED** [review]

File: receiver/statsdreceiver/factory.go [hunk 1]
```diff
@@ -23,13 +23,14 @@ import (
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/receiver"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/pprofextension/internal/metadata"
```
Comment (dmitryax) [inline receiver/statsdreceiver/factory.go:26]: Wrong package

**Review by dmitryax — CHANGES_REQUESTED** [review]

**Review by atoulme — APPROVED** [review]

**Review by altuner — APPROVED** [review]

**Comment by atoulme** [comment]
@dmitryax please take a look again, this should be good to go

**Review by jmacd — APPROVED** [review]

File: receiver/statsdreceiver/metadata.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+type: statsdreceiver
+
+status:
+  class: receiver
+  stability:
+    beta: [metrics]
+  distributions: [contrib, splunk]
```
Comment (codeboten) [inline receiver/statsdreceiver/metadata.yaml:7]: Needs another rebase, `sumo` was added. Sorry for all the churn @my-git9

**Review by codeboten — COMMENTED** [review]
Please update the factory to use the generated `Type` and `Stability` consts here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/96bed57799d082b6bb2c6f58d3996cca451c7d66/receiver/statsdreceiver/factory.go#L20-L21

File: receiver/statsdreceiver/metadata.yaml (hunk 2 above)
Comment (atoulme) [reply receiver/statsdreceiver/metadata.yaml:7]: I fixed the conflict with the github UI, lemme know if that works.

**Review by codeboten — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

File: receiver/statsdreceiver/metadata.yaml (hunk 2 above)
Comment (dmitryax) [reply receiver/statsdreceiver/metadata.yaml:7]: This is still unresolved. sumo is added in the doc only, not here

Comment (atoulme) [reply receiver/statsdreceiver/metadata.yaml]: I pushed 2 more commits to get this synchronized

**Pull request merged**